rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMService.java

```diff
@@ -141,13 +141,25 @@ String createNodeSource(String sessionId, String nodeSourceName, String infrastr
      * Deploys a node source and starts acquiring its nodes
      *
      * @param sessionId current session
-     * @param nodeSourceName name of the new NS
+     * @param nodeSourceName name of the node source to deploy
     
      * @throws RestServerException
      * @throws ServiceException
      */
     String deployNodeSource(String sessionId, String nodeSourceName) throws RestServerException, ServiceException;
 
+    /**
+     * Undeploys a node source and removes its nodes
+     *
+     * @param sessionId current session
+     * @param nodeSourceName name of the node source to undeploy
+    
+     * @throws RestServerException
+     * @throws ServiceException
+     */
+    String undeployNodeSource(String sessionId, String nodeSourceName, boolean force)
+            throws RestServerException, ServiceException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)